components/elm/src/biogeochem/FireMod.F90

```diff
@@ -372,7 +370,7 @@ subroutine FireArea (bounds, &
 
               ! For non-crop -- natural vegetation and bare-soil
               if( veg_pp%itype(p)  <  nc3crop .and. cropf_col(c)  <  1.0_r8 )then
-                 if( .not. shr_infnan_isnan(btran2(p))) then
+                 if( .not. (btran2(p) .ne. btran2(p)))then !?shr_infnan_isnan(btran2(p))) then
```

Comment:
I can't figure out his change equivalent to the original code.

Reply:
As far as I know, support for functions like is_nan() is a work-in-progess for gpu device code. This should be a platform independent way to check for NaNs.

(x .ne . x) will only ever be true for a NaN.  and so should be equivalent to shr_infnan_isnan(x).  And then in this case, it is negated to get only not NaNs.     I think this is currently the only instance where NaNs are checked for in the elm_run code